astropy/io/misc/asdf/connect.py

```diff
@@ -107,6 +107,12 @@ def asdf_identify(origin, filepath, fileobj, *args, **kwargs):
     return filepath is not None and filepath.endswith('.asdf')
 
 
-io_registry.register_reader('asdf', Table, read_table)
-io_registry.register_writer('asdf', Table, write_table)
-io_registry.register_identifier('asdf', Table, asdf_identify)
+try:
+    io_registry.register_reader('asdf', Table, read_table)
+    io_registry.register_writer('asdf', Table, write_table)
+    io_registry.register_identifier('asdf', Table, asdf_identify)
+except io_registry.IORegistryError:
+    # This means that the asdf-astropy version of thie connector
```

Comment:
```suggestion
    # This means that the asdf-astropy version of this connector
```

Reply:
Do we need to register at all? Why not just rely completely on `asdf-astropy` to do this for us?

Reply:
If asdf-astropy isn't installed, we want to fall back on the existing code, right?  Otherwise our deprecation will remove a feature that used to work.

Reply:
Then why not have the logic reflect that? Check to see if asdf-astropy is there and only try to run that block if not?